test/test_rb.py

```diff
@@ -0,0 +1,237 @@
+            self.assertAlmostEqual(
+                process_fidelity(Clifford(qc).to_matrix(), np.identity(2 ** num_qubits)),
+                1,
+                "Transpiled circuit doesn't result in the identity operator.",
+            )
+
+    def validate_metadata(self, circuits: list, exp_attributes: dict):
+        """
+        Validate the fields in "metadata" for the experiment.
+        Args:
+            circuits (list): A list containing quantum circuits
+            exp_attributes (dict): A dictionary with the experiment variable ands values
+        """
+        for ind, qc in enumerate(circuits):
+            self.assertEqual(
+                qc.metadata["xval"],
+                exp_attributes["lengths"][ind],
+                "The number of gates in the experiment metadata doesn't match to the one provided.",
+            )
+            self.assertEqual(
+                qc.metadata["qubits"],
+                tuple(exp_attributes["qubits"]),
+                "The qubits indices in the experiment metadata doesn't match to the one provided.",
+            )
+
+    def validate_circuit_data(
+        self, experiment: qe.experiment_data.ExperimentData, exp_attributes: dict
+    ):
+        """
+        Validate that the metadata of the experiment after it had run matches the one provided.
+        Args:
+
+            experiment(qiskit_experiments.experiment_data.ExperimentData): The experiment
+            data and results after it run.
+            exp_attributes (dict): A dictionary with the experiment variable ands values
+        """
+        for ind, data in enumerate(experiment.data):
+            experiment_information = data["metadata"]
+            self.assertEqual(
+                experiment_information["xdata"],
+                exp_attributes["lengths"][ind],
+                "The number of gates in the experiment doesn't match to the one in the metadata.",
+            )
+            self.assertEqual(
+                experiment_information["qubits"],
+                exp_attributes["qubits"],
+                "The qubits indices in the experiment doesn't match to the one in the metadata.",
+            )
+
+    def _exp_data_prpeties(self):
+        """
+        Return a list of dictionaries that contains invalid experiment propeties to check errors.
+        """
+        exp_data_list = [
+            {"qubits": [3, 3], "lengths": [1, 3, 5, 7, 9], "num_samples": 0, "seed": 100},
+            {"qubits": [-1], "lengths": [1, 3, 5, 7, 9], "num_samples": 1, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, -7, 9], "num_samples": 1, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, 7, 9], "num_samples": -4, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, 7, 9], "num_samples": 0, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 5, 5, 5, 9], "num_samples": 0, "seed": 100},
+        ]
+        return exp_data_list
+
+    def _test_input(self):
```

Comment:
@gadial - what do you think of this test?

Reply:
The test is ok, we need more of those. The problem is I don't understand what is wrong with the input here - should be explained in the comments.

Reply:
Added comment what is the input in each dictionary that raise for the error.